samples/AzureStorageQueuesPubSubSample/AzureStorageQueuesPubSub.cs

```diff
@@ -0,0 +1,141 @@
+using Azure.Storage.Queues;
+using Dapr.PluggableComponents.Components;
+using Dapr.PluggableComponents.Components.PubSub;
+
+internal sealed class AzureStorageQueuesPubSub : IPubSub
+{
+    private readonly ILogger<AzureStorageQueuesPubSub> logger;
+
+    private string? connectionString;
+    private string? queueName;
+    private TimeSpan pollInterval = TimeSpan.FromSeconds(5);
+    private int maxMessages = 5; 
+
+    public AzureStorageQueuesPubSub(ILogger<AzureStorageQueuesPubSub> logger)
+    {
+        this.logger = logger ?? throw new ArgumentNullException(nameof(logger));
+    }
+
+    #region IPubSub Members
+
+    public Task InitAsync(MetadataRequest request, CancellationToken cancellationToken = default)
+    {
+        this.connectionString = request.Properties["connectionString"];
+        this.queueName = request.Properties["queueName"];
+
+        if (request.Properties.TryGetValue("pollInterval", out var pollIntervalString))
+        {
+            this.pollInterval = TimeSpan.FromSeconds(Int32.Parse(pollIntervalString));
+        }
+
+        if (request.Properties.TryGetValue("maxMessages", out var maxMessagesString))
+        {
+            this.maxMessages = Int32.Parse(maxMessagesString);
+        }
+
+        return Task.CompletedTask;
+    }
+
+    public async Task PublishAsync(PubSubPublishRequest request, CancellationToken cancellationToken = default)
+    {
+        var queueClient = new QueueClient(this.connectionString, this.queueName);
+
+        await queueClient.CreateIfNotExistsAsync(cancellationToken: cancellationToken);
+
+        await queueClient.SendMessageAsync(Encoding.UTF8.GetString(request.Data.Span), cancellationToken);
+    }
+
+    public async Task PullMessagesAsync(PubSubPullMessagesTopic topic, IAsyncEnumerable<PubSubPullMessagesRequest> requests, IAsyncObserver<PubSubPullMessagesResponse> responses, CancellationToken cancellationToken = default)
+    {
+        var queueClient = new QueueClient(this.connectionString, this.queueName);
+
+        var acknowledgeTask =
+            async () =>
+            {
+                await foreach (var request in requests)
+                {
+                    if (request.AckMessageId != null)
```

Comment:
that way we are exposing the ack Id for the pubsub component, do we need it?

For the go-sdk implementation it wasn't necessary. see my comment [here](https://github.com/dapr/dapr/issues/5726#issuecomment-1376170894)

For the go/native implementation the way is much simpler though, the pubsub component receives a function as a parameter which is used as a handler when a new message arrives, that's all, I feel it is much easier than dealing with async enumerators,

However my main concern here is exposing the `ackId` for the component. An alternative is to embed a "Ack" function on each message that you sent and the component adaptor is responsaible for calling the ack function whenever the ack is returned by the runtime.


e.g

```csharp
new PubSubPullMessagesResponse {
...

    ackFunction = () => {
        await queueClient.DeleteMessageAsync(messageId, popReceipt, cancellationToken)
    }
}
```

Reply:
This is a good question; if we assume a base type is keeping the state needed to manage correlating incoming acknowledgements with outgoing messages, then we could provide a more integrated API.  The downside to the generated gRPC clients on .NET is that they aren't stateful, so it would shift that state to the component implementation, which would imply a base implementation type.  I'd still insist on a raw interface (as it looks now) to allow implementations that *don't* want such managed correlation the flexibility to do things "the harder way" if they choose.

Whether it would be "simpler" than the current approach is another question; I'll look and see what such a implementation might look like.

Reply:
Actually, I take part of that back; while the gRPC adaptor type isn't stateful as a whole, each `PullMessagesAsync()` call is a long-lived connection where the adaptor *could* maintain state.

Reply:
I've updated the`PullMessagesAsync()` API to eliminate the need for the pluggable component itself to manage IDs, using an approach similar to what you've suggested. It's elegant, for sure, but I'd say *just* on the edge of being a little complex for a .NET API (i.e. being passed a function that expects a function).